compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1039,7 +1039,10 @@ impl<'ast, 'ra: 'ast, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
                         .decl()
                         .inputs
                         .get(0)
-                        .map(|p| (p.span.shrink_to_lo(), "&self, "))
+                        .map(|p| {
+                            let suggestion = if p.is_reference() { "self" } else { "&self" };
```

Comment:
Note these aren't equivalent - the original is `"&self, "` (comma + space), the others are `"self"` / `"&self"`

Reply:
updated that part but p.<> is still giving error , will have to check where params is referenced or what type it is